app/uk/gov/hmrc/serviceconfigs/persistence/DeploymentConfigSnapshotRepository.scala

```diff
@@ -61,4 +64,18 @@ class DeploymentConfigSnapshotRepository @Inject()(mongoComponent: MongoComponen
       .deleteMany(new BasicDBObject())
       .toFuture
       .map(_ => ())
+
+  def populate(date: Instant): Future[Unit] = {
+
+    def forEnvironment(environment: Environment): Future[Unit] = {
+      for {
+        deploymentConfigs <- deploymentConfigRepository.findAll(environment)
+        snapshots         =  deploymentConfigs.map(DeploymentConfigSnapshot(date, _))
+        _                 <- collection.insertMany(snapshots).toFuture()
+      } yield ()
+    }
+
+    Environment.values
+      .foldLeft(Future.successful(()))((fUnit, env) => fUnit.flatMap(_ => forEnvironment(env)))
```

Comment:
you can use `Future.unit` instead of `Future.successful(())`
and `foldLeftM` instead of `foldLeft` to avoid the inner flatMap

Reply:
Nice one 👍